From 835d7c39e7075204ab96afb26c28752476732497 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 3 Sep 2021 18:00:32 +0200
Subject: [PATCH 133/841] feat(stm32mp2-fdts): add nvmem_layout OTP config

Add part number OTP, and NAND OTP words in NVMEM layout.
Part number OTP is OTP9.
The 2 OTP words used for NAND configs are OTP16 and OTP18.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 16bbe286720c2717e35a9bfea688c0f67f428f62 feat(stm32mp2-fdts): add nvmem_layout OTP config]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Change-Id: Ibba7270218e665fd8a2958fc0cee79b0a5a1aeb4
---
 fdts/stm32mp251.dtsi | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/fdts/stm32mp251.dtsi b/fdts/stm32mp251.dtsi
index f83730c9f..2c4f473d3 100644
--- a/fdts/stm32mp251.dtsi
+++ b/fdts/stm32mp251.dtsi
@@ -143,6 +143,23 @@
 			status = "disabled";
 		};
 
+		bsec: efuse@44000000 {
+			compatible = "st,stm32mp25-bsec";
+			reg = <0x44000000 0x400>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			part_number_otp: otp9@24 {
+				reg = <0x24 0x4>;
+			};
+			nand_otp: otp16@40 {
+				reg = <0x40 0x4>;
+			};
+			nand2_otp: otp20@50 {
+				reg = <0x50 0x4>;
+			};
+		};
+
 		rcc: rcc@44200000 {
 			compatible = "st,stm32mp25-rcc";
 			reg = <0x44200000 0x10000>;
-- 
2.47.2

